README.md
<img width="437" alt="Capture d’écran 2023-06-09 à 11 24 39" src="https://github.com/vengamo/vengamo-chat-ui/assets/133201077/a0b69b4b-9e0b-4a98-a313-0951676ab98e">


Vengamo Chat UI is a Flutter project that combines the user interface elements of WhatsApp, Telegram, and Facebook Messenger in one place. It offers a seamless and intuitive chat experience inspired by these popular messaging applications.

## Features

- Familiar UI elements from WhatsApp, Telegram, and Facebook Messenger.
- Customizable themes to personalize the chat interface.
- Image file support
- Audio files support 

## What's coming next
- Video file Support

## Early Feature request
 
 - [Get in touch](https://vengamo.com)
 - Send us an email vengamosocial@gmail.com
 - [Send us a message on whatsApp](https://api.whatsapp.com/send/?phone=33753861046&text=I+would+like+to+inquire+something+&type=phone_number&app_absent=0)
 - [Send us a message on Skype](https://join.skype.com/invite/Fz78g1fENfop)


## Screen Shoot

![Untitled design (7)](https://github.com/vengamo/vengamo-chat-ui/assets/133201077/acad32b0-5c4d-476a-9759-688498d73ba2)


## VengamoChatUI Drop in Properties.

- [Read documentation](https://pub.dev/documentation/vengamo_chat_ui/latest/vengamo_chat_ui/VengamoChatUI-class.html)

## Examples

```bash
   VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: false,
              time: getTime(),
              isAudio: true,
              audioSource: 'http://www.uscis.gov/files/nativedocuments/Track%2093.mp3',
              timeLabelColor: AppColors.iconColor,
              pointer: true,
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
   VengamoChatUI(
            senderBgColor: AppColors.softGreenColor,
            receiverBgColor: AppColors.white,
            isSender: true,
            isNextMessageFromSameSender: false,
            time: '10:00',
            timeLabelColor : AppColors.softBlackcolor,
            text: "Yolla ✋ Hey, guess what?",
            pointer: true, 
            ack: Icon(
                    Icons.check,
                    color: AppColors.iconColor, // You can customize the color here
                  size: 13, // You can customize the size here
            ),
          ),
        VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              timeLabelColor: AppColors.darkModeBackgroundColor,
              isSender: true,
              isNextMessageFromSameSender: false,
              imgUrl : 'https://wallpaperaccess.com/full/1248267.jpg',
              caption: "Why did the sun never want to join the galaxy's talent show? Because it didn't want to be a star performer, it preferred to shine solo",
              time: '07:05',
              pointer: true, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
   )
   ```

## Getting Started

To use Vengamo Chat UI in your Flutter project, follow these steps:

1. Ensure that you have Flutter installed on your system. For installation instructions, refer to the [Flutter official documentation](https://flutter.dev/docs/get-started/install).

2. Open your project's `pubspec.yaml` file.

3. Add the following dependency under the `dependencies` section:

   ```yaml
   dependencies:
     vengamo_chat_ui: ^1.2.5
   ```

   Replace `^1.2.5` with the desired version of Vengamo Chat UI.

4. Save the `pubspec.yaml` file.

5. Run the following command in your project's root directory:

   ```bash
   flutter pub get
   ```

6. Import Vengamo Chat UI in your Dart code:

   ```dart
   import 'package:vengamo_chat_ui/vengamo_chat_ui.dart';
   ```

7. You can now use Vengamo Chat UI components in your Flutter app.

## Project maintainer

<img src="https://github.com/vengamo/vengamo-chat-ui/assets/133201077/7030deee-4d00-4de9-83d0-e0e46800070b" alt="Profile Image" height="45" width="45">
 
.[Eric Weeb](https://github.com/vengamo)


## Contributions

Contributions to the Vengamo Chat UI project are welcome! If you find any issues or have suggestions for improvements, feel free to open an issue or submit a pull request. Please follow the project's code of conduct.

## License

The Vengamo Chat UI project is licensed under the [MIT License](LICENSE).

Feel free to use this README template as a starting point and add more details as needed to provide a comprehensive overview of your Vengamo Chat UI project.

CHANGELOG.md
## 1.2.6
- Updated example page
## 1.2.5
- Added Audio support
  . Files cashing for fast loading
  . Improved image load (x3 faster than before)
## 1.2.4
- Updated base64 http request simulator
## 1.2.3
- Updated Example page
## 1.2.2
- Image loading place holder and call back function
## 1.2.1
- Minor change & Image optimization
## 1.2.0
- Updated Image support
  .imgUrl  can be empty String or null
## 1.1.9
- Support of Base 64 String 
   .You can now use base 64 string as imgUrl
## 1.1.8
- Removed google fonts
## 1.1.7
- Added loading progress bar animation while Image is being uploading or downloading
## 1.1.6
- Added Image caption
## 1.1.5
- Updated example
## 1.1.4
- Updated Image file custom background color
## 1.1.3
- Updated Example.
## 1.1.2
- Added image file support.
## 1.1.1
- Updated Readme file.
## 1.1.0
- Updated example.
## 1.0.9
- updated chat bubble for both sender and receiver

## 1.0.8
- added customizable chat bubble
   .Text colors, status Message( sent, delivered and seen)
   .chat bubble background color

## 1.0.7
- added pointer, chat bubble colors & message status
## 1.0.6
- Added Github repo & home page

## 1.0.5
- Added logo

## 1.0.4
- Added custom background color support

## 1.0.3
- Added types

## 1.0.2

- Added example

## 1.0.1

- Initial release


lib/main.dart
import 'package:flutter/material.dart';
import 'package:vengamo_chat_ui/vengamo_chat_ui.dart';
import 'package:vengamo_chat_ui/theme/app_color.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'James Wagon',
      theme: ThemeData(
        primarySwatch: Colors.blueGrey,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: const VengamoChatScreen(),
    );
  }
}

class VengamoChatScreen extends StatelessWidget {
  const VengamoChatScreen({super.key});

  String getTime() {
    int hour = DateTime.now().hour;
    int min = DateTime.now().minute;
    String hRes = hour <= 9 ? '0$hour' : hour.toString();
    String mRes = min <= 9 ? '0$min' : min.toString();
    return '$hRes:$mRes';
  }


  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('James Wagon'),
      ),
      body: Container(
        width: MediaQuery.of(context).size.width,
        height: MediaQuery.of(context).size.height,
        decoration: const BoxDecoration(
              color: AppColors.backgroundColor,
            ),
 
        child: SingleChildScrollView(
          child: Padding(
            padding: const EdgeInsets.only(left: 20, right: 20, bottom: 20, top: 2),
            child: Column(
              children: [

              VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: false,
              time: getTime(),
              isAudio: true,
              audioSource: 'http://www.uscis.gov/files/nativedocuments/Track%2093.mp3',
              timeLabelColor: AppColors.iconColor,
              pointer: true,
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
            const SizedBox(height: 10,),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: false,
              isNextMessageFromSameSender: false,
              time: getTime(),
              isAudio: true,
              audioSource: 'https://p.scdn.co/mp3-preview/c2bca95698cc381f0a1a9111095156d15ebd4698',
              timeLabelColor: AppColors.iconColor,
              pointer: true,
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
            // const SizedBox(height: 10,),
            //     VengamoChatUI(
            //   senderBgColor: AppColors.softGreenColor,
            //   receiverBgColor: AppColors.white,
            //   timeLabelColor: AppColors.darkModeBackgroundColor,
            //   isSender: false,
            //   isNextMessageFromSameSender: false,
            //   imgUrl : 'https://static.wikia.nocookie.net/frstarwars/images/4/43/TheChild-Fathead.png/revision/latest?cb=20201101090715',
            //   caption : "check this out! Why don't skeletons fight each other? They don't have the guts!",
            //   time: getTime(),
            //   pointer: true, 
            //   ack: Image.asset(
            //     'assets/images/seen.png',
            //     height: 12,
            //     width: 12,
            //   ),
            // ),
            const SizedBox(height: 10,),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: false,
              timeLabelColor: AppColors.white,
              isNextMessageFromSameSender: false,
              imgUrl : 'https://cdn.wallpapersafari.com/28/2/vrIzJD.jpg',
              time: getTime(),
              pointer: true, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
            const SizedBox(height: 10,),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: false,
              timeLabelColor: AppColors.white,
              isNextMessageFromSameSender: false,
              imgUrl : 'https://weellu.s3.us-east-2.amazonaws.com/test/kYFhiApyyVRX.jpeg',
              time: getTime(),
              pointer: true, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
               VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: false,
              time: getTime(),
              timeLabelColor : AppColors.softBlackcolor,
              text: "Yolla ✋ Hey, guess what?",
              pointer: true, 
              ack: const Icon(
                      Icons.check,
                      color: AppColors.iconColor, // You can customize the color here
                    size: 13, // You can customize the size here
              ),
            ),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: true,
              time: getTime(),
              timeLabelColor : AppColors.softBlackcolor,
              text: "What?",
              pointer: false, 
              ack: Image.asset(
                'assets/images/double_tick_grey.png',
                height: 13,
                width: 13,
              ),
            ),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: false,
              isNextMessageFromSameSender: false,
              time: getTime(),
              text: "I invented a new word!.",
              pointer: true, 
              ack: Image.asset(
                'assets/images/double_tick_green.png',
                height: 14,
                width: 15,
              ),
            ),
            const SizedBox(height: 5,),

            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              timeLabelColor: AppColors.darkModeBackgroundColor,
              isSender: true,
              isNextMessageFromSameSender: false,
              imgUrl : 'https://wallpaperaccess.com/full/1248267.jpg',
              caption: "Why did the sun never want to join the galaxy's talent show? Because it didn't want to be a star performer, it preferred to shine solo",
              time: getTime(),
              pointer: true, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: false,
              isNextMessageFromSameSender: true,
              time: getTime(),
              text: "Hold on, a sec, let me pick this call?",
              pointer: false, 
              ack: Image.asset(
                'assets/images/double_tick_green.png',
                height: 14,
                width: 15,
              ),
            ),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: false,
              isNextMessageFromSameSender: true,
              time: getTime(),
              text: "Plagiarism! 💫 😂 ",
              pointer: false, 
              ack: Image.asset(
                'assets/images/double_tick_green.png',
                height: 14,
                width: 15,
              ),
            ),
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: false,
              time: getTime(),
              text: "Plagiarism!? 😂.",
              pointer: true, 
              ack: Image.asset(
                'assets/images/sent.png',
                height: 12,
                width: 12,
              ),
            ),
           
             VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: true,
              time: getTime(),
              text: "Thanks! I thought you'd like it 🤪",
              pointer: false, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ), VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: true,
              time: getTime(),
              text: "Why don't scientists trust atoms? Because they make up everything! If you'd like to hear more jokes or have any other questions, feel free to ask.😂🤪",
              pointer: false, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
        
            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: false,
              isNextMessageFromSameSender: false,
              time: getTime(),
              text: "check this out! Why don't skeletons fight each other? They don't have the guts!",
              pointer: true, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),
            
            const SizedBox(height: 5,),

            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              isNextMessageFromSameSender: false,
              imgUrl : 'https://cache.marieclaire.fr/data/photo/w1000_ci/1ju/sean-connery-james-bond.jpg',
              caption: "Why did the sun never want to join the galaxy's talent show? Because it didn't want to be a star performer, it preferred to shine solo",
              time: getTime(),
              pointer: true, 
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),

            VengamoChatUI(
              senderBgColor: AppColors.softGreenColor,
              receiverBgColor: AppColors.white,
              isSender: true,
              caption: "You'd like it 🤪",
              isNextMessageFromSameSender: false,
              imgUrl : 'https://wallpapers.com/images/high/peaceful-relaxing-vhbesnvs3ylhpuvk.webp',
              time: getTime(),
              timeLabelColor: AppColors.iconColor,
              pointer: false,
              ack: Image.asset(
                'assets/images/seen.png',
                height: 12,
                width: 12,
              ),
            ),

           
        
              ],
            ),
          ),
        ),
      ), // Use Vengamo Chat UI component here
    );
  }
}